miss_islington/status_change.py

```diff
@@ -106,12 +107,9 @@ async def check_ci_status_and_approval(
                 )
                 success = result["state"] == "success" and not failure
                 if leave_comment:
-                    if failure:
+                    if failure or not success:
                         emoji = "❌"
                         status = "it's a failure or timed out"
-                    elif not success:
-                        emoji = "❌"
-                        status = "it's a failure"
```

Comment:
Was this change intentional?

@DanielNoord recently fixed this.

Reply:
@ambv Was this indeed intentional? If not I could open a PR to revert it to my last fix?

Reply:
@ezio-melotti I checked in `cpython` and it seems to be okay, although there are some weird comments that are probably due to checks timing out and then being rerun.

It doesn't regress on https://github.com/python/miss-islington/pull/576/ but it does remove the narrowing down that the `"it's a failure"` comment gives. Previously if that was the only thing in the comment you knew there wasn't a timeout, so there was likely an actual issue with the PR. You no longer get this information, but I'm not sure it warrants a PR to re-introduce that. It doesn't seem to helpful anyway.

That said, it could be simplified to `if not success`. If `failure == True` then `success` will always be `False`, so that doesn't need double-checking. Want me to open a PR for that?

Reply:
In https://github.com/python/cpython/pull/97820#issuecomment-1266315714 I got the `Status check is done, and it's a failure or timed out ❌.` message, even though there was no evident failure.  https://github.com/python/cpython/pull/97821#issuecomment-1266320540 reported success.

Reply:
It's hard to debug these comments after the fact.
https://api.github.com/repos/python/cpython/commits/66eca5d2f185cd6943afeb69dae3eba625bbd077/status
https://api.github.com/repos/python/cpython/commits/66eca5d2f185cd6943afeb69dae3eba625bbd077/check-runs

Both show that there are no failed runs. So I don't know what is going on there honestly... But I'm also fairly sure it isn't caused by the changes in this PR.

Reply:
I cannot find it anymore but I had an example where it **was** a timeout but TIMED_OUT was not set so the bot would report a failure. And since the best the message could say anyway was that it's *either* a failure *or* a timeout, I decided to just shorten it.

IIRC this change also let me not touch more tests, which otherwise I would have to alter due to the `leave_comment=False` change.

What's more worrying is that @ezio-melotti's example are still the bot leaving comments even those `leave_comment=False` is set. What gives?

Reply:
Turns out this was due to a missing change that I now also cover in https://github.com/python/miss-islington/pull/583

Reply:
Also, I remember what I meant by "this change also let me not touch more tests". The change in this PR caused test coverage to fall due to one of those branches to not be exercised anymore.